### PR TITLE
Add GitHub Actions

### PR DESCRIPTION
**Pull request by Scrumplex**
Let's use GitHub Actions for a few things. Namely let's provide binaries for new pushes and PRs. That way everyone can test a specific revision or PR, without needing to build it themselves.

TODOs:
- [ ] Add Windows build
- [x] (Optional) Add macOS build
- [ ] ~~Add other Linux distros~~
  - [ ] ~~(Optional) Arch Linux~~
  - [ ] ~~(Optional) Fedora~~
  - [ ] ~~(Optional) openSUSE~~

### CODE REVIEW

**Comment by Scrumplex** [comment]
Looks like this generates quite a few emails. I am gonna work on a different branch while testing

**Comment by Scrumplex** [comment]
Other Linux distros aren't natively supported by GitHub actions. Would probably need to do something with Docker. It isn't high priority currently.

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+      - name: ccache
+        uses: hendrikmuhs/ccache-action@v1
+        with:
+          key: ${{ matrix.os }}-${{ runner.arch }}
+          max-size: 100M
+
+      - name: Cache Qt
+        id: cache-qt
+        uses: actions/cache@v1  # not v2! TODO: why though?
```
Comment (ghost) [inline .github/workflows/build.yaml:27]: >

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      - name: Setup cmake
+        uses: jwlawson/actions-setup-cmake@v1.12
+        with:
+          cmake-version: '3.22.x'
+
+      - name: Build
+        run: |
+          export PATH="/usr/lib/ccache:/usr/local/opt/ccache/libexec:$PATH"
+          mkdir -p build install
+          cmake -B build -S . -DCMAKE_INSTALL_PREFIX=install
```
Comment (ghost) [inline .github/workflows/build.yaml]: Why not `/usr` as the prefix?

You can install to `install` directory with `DESTDIR=install` in the make command,

**Review by ghost — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yaml (hunk 2 above)
Comment (Scrumplex) [reply .github/workflows/build.yaml]: I used the example from [BUILD.md](https://github.com/PolyMC/PolyMC/blob/068094583912797855cc7ef8045198d51eedcfac/BUILD.md#building-from-command-line)

File: .github/workflows/build.yaml (hunk 1 above)
Comment (Scrumplex) [reply .github/workflows/build.yaml:27]: I used the example from: [install-qt-action](https://github.com/jurplel/install-qt-action#cached). It isn't explained why not v2

File: .github/workflows/build.yaml (hunk 2 above)
Comment (ghost) [reply .github/workflows/build.yaml]: That's for lin-nodeps only, do you want to build with that?

Comment (Scrumplex) [reply .github/workflows/build.yaml]: Yeah. I think the idea was just to have portable builds for PRs. That way anyone (currently only Linux and macOS) can test PRs with ease

**Comment by ghost** [comment]
Looks good. Given the evident difficulty of making a Windows build on here, there is now a specific Jenkins project that will build the currently-focused PR on Windows: https://jenkins.polymc.org/job/PolyMC-PRs/

**Comment by Scrumplex** [comment]
Succeeded by #122

**Pull request closed without merging**